sdk/pinocchio/src/cpi.rs

```diff
@@ -184,11 +251,103 @@ pub unsafe fn invoke_signed_unchecked(
     instruction: &Instruction,
     accounts: &[Account],
     signers_seeds: &[Signer],
+) {
+    invoke_instruction_signed_unchecked(
+        instruction.program_id,
+        instruction.accounts,
+        instruction.data,
+        accounts,
+        signers_seeds,
+    );
+}
+
+/// Invoke a cross-program instruction.
+#[inline(always)]
+pub fn invoke_instruction<const ACCOUNTS: usize>(
+    program_id: &Pubkey,
+    priviledges: &[Priviledge; ACCOUNTS],
+    instruction_data: &[u8],
+    accounts: &[Account; ACCOUNTS],
+) -> ProgramResult {
+    invoke_instruction_signed(program_id, priviledges, instruction_data, accounts, &[])
+}
+
+/// Invoke a cross-program instruction with signatures.
+#[inline]
+pub fn invoke_instruction_signed<const ACCOUNTS: usize>(
+    program_id: &Pubkey,
+    priviledges: &[Priviledge; ACCOUNTS],
+    instruction_data: &[u8],
+    accounts: &[Account; ACCOUNTS],
+    signers_seeds: &[Signer],
+) -> ProgramResult {
```

Comment:
I see what you're trying to do here, but the API is a bit strange, since it's requiring the `privileges` line up perfectly with `accounts`. Are there a lot of gains from it? Because it seems a bit delicate to use.

Reply:
Yes, since we avoid the `Pubkey` comparison to align with each `AccountInfo`, as the `invoke_signed` requires. But maybe the solution is for clients to optimize this and use `invoke_signed_unchecked`.

Reply:
I have done a bit more testing with the invoke_signed and it seems that there is almost no difference compared to the proposed invoke_instruction_signed. I think it is ok to remove the new ones and keep the existing API.